cpp/src/cider-velox/benchmark/expression/StringFunctionBenchmark.cpp

```diff
@@ -0,0 +1,221 @@
+#include "velox/functions/prestosql/Comparisons.h"
+#include "velox/substrait/VeloxToSubstraitPlan.h"
+#include "velox/vector/arrow/Bridge.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+// #include "velox/functions/Udf.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+
+// This file refers velox/velox/benchmarks/basic/SimpleArithmetic.cpp
+DEFINE_int64(fuzzer_seed, 99887766, "Seed for random input dataset generator");
+DEFINE_int64(batch_size, 1000, "batch size for one loop");
+DEFINE_int64(loop_count, 100'000, "loop count for benchmark");
+DEFINE_bool(dump_ir, false, "dump llvm ir");
+
+using namespace cider::exec::processor;
+using namespace cider::exec::nextgen::context;
+using namespace facebook::velox;
+using namespace facebook::velox::memory;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::test;
+using namespace facebook::velox::plugin;
+using namespace facebook::velox::functions;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::substrait;
+
+// static const std::shared_ptr<PoolAllocator> allocator =
+//     std::make_shared<CiderDefaultAllocator>();
+namespace {
+std::pair<ArrowArray*, ArrowSchema*> veloxVectorToArrow(RowVectorPtr vec,
+                                                        MemoryPool* pool) {
+  for (size_t i = 0; i < vec->childrenSize(); i++) {
+    vec->childAt(i)->mutableRawNulls();
+  }
+  ArrowArray* inputArrowArray = CiderBatchUtils::allocateArrowArray();
+  exportToArrow(vec, *inputArrowArray, pool);
+  ArrowSchema* inputArrowSchema = CiderBatchUtils::allocateArrowSchema();
+  exportToArrow(vec, *inputArrowSchema);
+
+  return {inputArrowArray, inputArrowSchema};
+}
+
+class StringFunctionBenchmark : public functions::test::FunctionBenchmarkBase {
+ public:
+  using ArrowArrayReleaser = void (*)(struct ArrowArray*);
+  explicit StringFunctionBenchmark(size_t vectorSize) : FunctionBenchmarkBase() {
+    functions::prestosql::registerStringFunctions();
+
+    inputType_ = ROW({{"col_str", VARCHAR()}, {"col_str2", VARCHAR()}});
+
+    vectorSize_ = vectorSize;
+    // Generate input data.
+    VectorFuzzer::Options opts;
+    opts.vectorSize = vectorSize;
+    // opts.nullRatio = 0.5;
+    // opts.stringVariableLength = false;
+    // opts.stringLength = 16;
+    VectorFuzzer fuzzer(opts, pool(), FLAGS_fuzzer_seed);
+
+    std::vector<VectorPtr> children;
+    children.emplace_back(fuzzer.fuzzFlat(VARCHAR()));  // varchar
+    children.emplace_back(fuzzer.fuzzFlat(VARCHAR()));  // varchar
+
+    rowVector_ = std::make_shared<RowVector>(
+        pool(), inputType_, nullptr, vectorSize, std::move(children));
+
+    ArrowSchema* _schema;
+    std::tie(inputArray_, _schema) = veloxVectorToArrow(rowVector_, execCtx_.pool());
+    // hack: make processor don't release inputArray_, otherwise we can't use inputArray_
+    // multi times.
+    inputReleaser_ = inputArray_->release;
```

Comment:
So we use same batch multi times? will it affect benchmark result?

Reply:
yes. I think it won't.